python/cudf/cudf/core/dataframe.py

```diff
@@ -7639,7 +7639,7 @@ def _get_union_of_indices(indexes):
     else:
         merged_index = cudf.core.index.GenericIndex._concat(indexes)
         merged_index = merged_index.drop_duplicates()
-        _, inds = merged_index._values.sort_by_values()
+        inds = merged_index._values.argsort()
         return merged_index.take(inds)
```

Comment:
```suggestion
        return merged_index.take(inds, check_bounds=False)
```

Reply:
Although, am I missing something here? 

Why is

```python
foo.take(foo.argsort())
```
not the same as
```
foo.sort_values()
```
?

Reply:
Ah, `Index.sort_values` is just implemented as this anyway, so this is a future enhancement.